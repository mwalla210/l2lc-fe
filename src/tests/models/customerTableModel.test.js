import CustomerTableModel from '../../models/CustomerTableModel'
import renderer from 'react-test-renderer'

jest.mock('../../components/tableActionCell')
jest.mock('../../components/projectStatusCell')
jest.mock('../../components/projectStatusFilter')

jest.mock('../../api', () => {
  return {
    fetchCustomers: jest.fn()
  }
})
jest.mock('../../store/website', () => {
  return {
    setCustomer: jest.fn(),
    currentProject: {
      changeCustomer: jest.fn()
    },
    updateProject: jest.fn().mockReturnValue(Promise.resolve()),
    createProject: jest.fn().mockReturnValue(Promise.resolve()),
  }
})

describe('CustomerTableModel', () => {
  it('Tests constructor', () => {
    let customer = new CustomerTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    expect(customer).toHaveProperty('columns')
})

  it('Tests clickHandler with info', () => {
    let customer = new CustomerTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    expect(customer.infoClickNav.mock.calls.length).toBe(0)
    customer.clickHandler(1,'info')
    expect(customer.infoClickNav.mock.calls.length).toBe(1)
  })

  it('Tests clickHandler with edit', () => {
    let customer = new CustomerTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    expect(customer.editClickNav.mock.calls.length).toBe(0)
    customer.clickHandler(1,'edit')
    expect(customer.editClickNav.mock.calls.length).toBe(1)
  })

  it('Tests clickHandler with no input', () => {
    let customer = new CustomerTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    const website = require('../../store/website')
    expect(website.setCustomer.mock.calls.length).toBe(2)
    customer.clickHandler(1,'else')
    expect(website.setCustomer.mock.calls.length).toBe(2)
  })

  it('Tests actionColumns', () => {
    let customer = new CustomerTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    customer.actionColumns()
    expect(customer.columns[5].getProps()).toEqual({className: 'center',style: {paddingTop: '0px',paddingBottom: '0px'}})
    const component = renderer.create(customer.columns[5].Cell())
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Tests noActionColumns', () => {
    let customer = new CustomerTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    customer.noActionColumns()
    const component = renderer.create(customer.columns[5].Cell({}))
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Tests selectCreateClick', () => {
    let customer = new CustomerTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    expect(customer.selectNav.mock.calls.length).toBe(0)
    customer.selectCreateClick(1)
    expect(customer.selectNav.mock.calls.length).toBe(0)
  })

  it('Tests selectUpdateClick', () => {
    let customer = new CustomerTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    expect(customer.selectNav.mock.calls.length).toBe(0)
    customer.selectUpdateClick(1)
    expect(customer.selectNav.mock.calls.length).toBe(0)
  })

  it('Tests selectCreateTable', () => {
    let customer = new CustomerTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    customer.noActionColumns = jest.fn()
    expect(customer.noActionColumns.mock.calls.length).toBe(0)
    customer.selectCreateTable()
    expect(customer.noActionColumns.mock.calls.length).toBe(1)
  })

  it('Tests selectUpdateTable', () => {
    let customer = new CustomerTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    customer.noActionColumns = jest.fn()
    expect(customer.noActionColumns.mock.calls.length).toBe(0)
    customer.selectUpdateTable()
    expect(customer.noActionColumns.mock.calls.length).toBe(1)
  })

  it('Tests nonSelectTable', () => {
    let customer = new CustomerTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    customer.actionColumns = jest.fn()
    expect(customer.actionColumns.mock.calls.length).toBe(0)
    customer.nonSelectTable()
    expect(customer.actionColumns.mock.calls.length).toBe(1)
  })

})
