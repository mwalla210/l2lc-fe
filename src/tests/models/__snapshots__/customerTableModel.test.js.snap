// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerTableModel Tests actionColumns 1`] = `
<span>
  <span>
    <button
      aria-label="Left Align"
      className="btn-circle btn btn-outline-secondary"
      disabled={undefined}
      onClick={[Function]}
      type="button"
    >
      <img
        alt="info"
        src="../../style/open-iconic-master/svg/info.svg"
        style={undefined}
      />
    </button>
  </span>
</span>
`;

exports[`CustomerTableModel Tests noActionColumns 1`] = `
<span>
  <span>
    <button
      className="btn btn-info"
      disabled={undefined}
      onClick={[Function]}
      style={undefined}
      type="button"
    >
      Select
    </button>
  </span>
</span>
`;

exports[`CustomerTableModel Tests non-admin actionColumns 1`] = `
<span>
  <span>
    <button
      aria-label="Left Align"
      className="btn-circle btn btn-outline-secondary"
      disabled={undefined}
      onClick={[Function]}
      type="button"
    >
      <img
        alt="info"
        src="../../style/open-iconic-master/svg/info.svg"
        style={undefined}
      />
    </button>
    <button
      aria-label="Left Align"
      className="btn-circle btn btn-outline-secondary"
      disabled={undefined}
      onClick={[Function]}
      type="button"
    >
      <img
        alt="pencil"
        src="../../style/open-iconic-master/svg/pencil.svg"
        style={
          Object {
            "marginLeft": "2px",
          }
        }
      />
    </button>
  </span>
</span>
`;
